test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -420,35 +420,64 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 		})
 	})
 
-	// TODO: we have a dependency between tests(that in general bad practice, but saves us some tests run time),
-	// once we will want to run tests in the random order or without failFast we will need to refactor tests
 	Context("Updating of nodeSelector parameter and node labels", func() {
 		var mcp *machineconfigv1.MachineConfigPool
 		var newCnfNode *corev1.Node
 
 		newRole := "worker-test"
 		newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+		var mcpSelectorRemoved, labelsDeletion = false, false
 		newNodeSelector := map[string]string{newLabel: ""}
 
-		testutils.BeforeAll(func() {
+		existingMcpSelector := map[string]string{
+			fmt.Sprintf("%s/%s", "machineconfiguration.openshift.io", "role"): testutils.RoleWorkerCNF,
+		}
+		var removeLabels = func(nodeSelector map[string]string, targetNode *corev1.Node) error {
```

Comment:
this works, but at glance it seems this code can be a regular function? IOW, why do we need to define this inline here?

Reply:
Because this code is very specific to this test case .Also having this function inline within this context, its easy to understand the code easily. This helps in having 2 test cases independent.

Reply:
Also this code is very specific to the test case of checking the configuration after the node labels are reverted.  so the steps are :
1. Remove the node labels 
2. check the configuration test 
3. Revert the node labels 
Since step1 is mostly a setup part , we want to put in Before Each. But at the same time we want this to be executed only if this test case is executed. So we do not want this code to be executed if we are not calling this test case. 
Also step3 we have put in AfterEach. and we want this to be executed only when Step1 happens. Hence we have the variable "labelsDeletion" which acts as a flag. 
Due to the above reasons , i believe the below code suits the above requirements.

Reply:
not a fan, but let's keep this direction for now.